scripts/automated_ingestion/eessitarball.py

```diff
@@ -156,6 +180,47 @@ def run_handler(self):
         handler = self.states[self.state]['handler']
         handler()
 
+    def verify_signatures(self):
+        """Verify the signatures of the downloaded tarball and metadata file using the corresponding signature files."""
+
+        sig_missing_msg = 'Signature file %s is missing.'
+        sig_missing = False
+        for sig_file in [self.local_sig_path, self.local_metadata_sig_path]:
+            if not os.path.exists(sig_file):
+                logging.warning(sig_missing_msg % sig_file)
+                sig_missing = True
+
+        if sig_missing:
+            # If signature files are missing, we return a failure,
+            # unless the configuration specifies that signatures are not required.
+            if self.config['signatures'].getboolean('signatures_required', True):
+                return False
+            else:
+                return True
+
+        verify_script = self.config['signatures']['signature_verification_script']
+        allowed_signers_file = self.config['signatures']['allowed_signers_file']
+        if not os.path.exists(verify_script):
+            logging.error(f'Unable to verify signatures, the specified signature verification script does not exist!')
+            return False
```

Comment:
If signature files are provided, but signatures are not required the procedure would return False if the verification script cannot be found. Maybe we would need to check here if signatures are required and only return False if the verification script cannot be found?

Reply:
Maybe the variable name `signatures_required` is a bit ambiguous, but my intention here was that if signatures are provided, they should always be checked and be correct, and hence there has to be a verification script. With `signatures_required = no` it's okay to not provide signature, and with `signatures_required` every tarball should have a signature file.

Reply:
Maybe a better name would be something like `allow_missing_signatures`?

Reply:
That or little comment? But I understand the intention, so would be fine to keep as is.

Reply:
Added a comment in https://github.com/EESSI/filesystem-layer/pull/208/commits/f7a60f6f05438bdd9e8c329222777b91985bf831.

Reply:
Nice